framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -795,7 +795,14 @@ void MediaPlayerImpl::playback()
 			}
 		}
 	} else if (num_read == 0) {
-		stop();
+		audio_manager_result_t result = stop_audio_stream_out();
+		if (result != AUDIO_MANAGER_SUCCESS) {
+			meddbg("stop_audio_stream_out failed ret : %d\n", result);
+		}
+
+		PlayerWorker &mpw = PlayerWorker::getWorker();
```

Comment:
How about moving this line repeatedly declared in line 785, 803, and 810 into line 779 or 780?